beta.go

```diff
@@ -0,0 +1,58 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mathext
+
+import (
+	"math"
+)
+
+// Beta returns the value of the complete beta function B(a, b).
+// B(a,b) is defined as: Γ(a)Γ(b) / Γ(a+b)
+// B(a,b) returns +Inf if a xor b is 0
+// B(a,b) returns  NaN if a  or b is < 0
+// B(a,b) returns  NaN if a  or b is NaN.
+//
+// See http://mathworld.wolfram.com/BetaFunction.html for more detailed informations.
+func Beta(a, b float64) float64 {
+	switch {
+	case (a == 0 && b > 0) || (a > 0 && b == 0):
+		return math.Inf(+1)
+	case a < 0 || b < 0:
```

Comment:
Add 0,0 in this switch. Then you can do
- If both == 0
- if either < 0
- if either nan
- if either == 0